src/viam/components/board/service.py

```diff
@@ -176,8 +176,12 @@ async def WriteAnalog(self, stream: Stream[WriteAnalogRequest, WriteAnalogRespon
         assert request is not None
         name = request.name
         board = self.get_resource(name)
+        try:
+            analog_writer = await board.analog_by_name(request.pin)
+        except ResourceNotFoundError as e:
+            raise e.grpc_error
         timeout = stream.deadline.time_remaining() if stream.deadline else None
-        await board.write_analog(pin=request.pin, value=request.value, timeout=timeout, metadata=stream.metadata)
+        await analog_writer.write(value=request.value, timeout=timeout, metadata=stream.metadata, extra=request.extra)
```

Comment:
You have to convert `request.extra` from a `Struct` to a `dict`: `struct_to_dict(request.extra)`

This will also get rid of the type checking error that's currently causing the build to fail.